File: itzel_be/src/main/java/com/itzel/servicio/contabilidad/ClasificadorService.java
package com.itzel.servicio.contabilidad;

import com.itzel.modelo.contabilidad.Clasificador;
import com.itzel.repositorio.contabilidad.ClasificadorR;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
@RequiredArgsConstructor
@Service
public class ClasificadorService {
    @Autowired
    private ClasificadorR dao;
    public List<Clasificador> findAll(){
        return dao.findAll();
    }
    public List<Clasificador> findByCodOrNom(String dato){
        return dao.findByCodOrNom(dato);
    }


    // Busca por codpar y nompar ordenado por codpar
    public List<Clasificador> buscarPorCodparYNompar(String codpar, String nompar) {
        return dao.findByCodparStartingWithAndNomparContainingIgnoreCaseOrderByCodparAsc(codpar, nompar);
    }

    // Valida codpar
    public boolean valCodpar(String codpar) {
        return dao.existsByCodpar(codpar);
    }

    // Validar nompar
    public boolean valNompar(String nompar) {
        return dao.existsByNomparIgnoreCase(nompar);
    }

    // Partidas de ingresos del clasificador para datalist
    public List<Clasificador> obtenerPartidasing(String codpar) {
        return dao.findPartidasing(codpar+'%');
    }

    // Guarda nuevo
    public <S extends Clasificador> S save(S entity) {
        return dao.save(entity);
    }

    // Busca por idclasificador
    public Optional<Clasificador> findById(Short idclasificador) {
        return dao.findById(idclasificador);
    }

    // Actualizar
    public Clasificador actualiza(Short idclasificador, Clasificador x) {
        Optional<Clasificador> y = dao.findById(idclasificador);
        if (y.isPresent()) {
            Clasificador partida = y.get();
            partida.setCodpar(x.getCodpar());
            partida.setNompar(x.getNompar());
            partida.setNivpar(x.getNivpar());
            partida.setGrupar(x.getGrupar());
            partida.setDespar(x.getDespar());
            partida.setCueejepresu(x.getCueejepresu());
            partida.setPresupuesto(x.getPresupuesto());
            partida.setEjecucion(x.getEjecucion());
            partida.setDevengado(x.getDevengado());
            partida.setReforma(x.getReforma());
            partida.setAsignaini(x.getAsignaini());
            partida.setGrupo(x.getGrupo());
            partida.setBalancostos(x.getBalancostos());

            partida.setUsumodi(x.getUsumodi());
            partida.setFecmodi(x.getFecmodi());

            return dao.save(partida);
        } else {
            throw new RuntimeException("Partida del clasificador no encontrada con id " + idclasificador);
        }
    }

    // Eliminar
    public void deleteById(Short idclasificador) {
        dao.deleteById(idclasificador);
    }

}

File: itzel2.1/src/app/componentes/servicios/contabilidad/clasificador.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../../environments/environment';
import { HttpClient } from '@angular/common/http';
import { Clasificador } from '../../modelos/contabilidad/clasificador.model';
import { Observable } from 'rxjs';


const apiUrl = environment.API_URL;
const baseUrl = `${apiUrl}/clasificador`;

@Injectable({
  providedIn: 'root'
})

export class ClasificadorService {

  constructor(private http: HttpClient) { }

  getAllClasificador() {
    return this.http.get(`${baseUrl}`);
  }
  getAllClasificadorByCodNom(dato: string) {
    return this.http.get(`${baseUrl}/codnompar?dato=${dato}`);
  }

  //Busca las partidas del clasificador codpar y nompar
  getByCodparyNompar(codpar: String, nompar: String) {
    return this.http.get<Clasificador[]>(`${baseUrl}/busca?codpar=${codpar}&nompar=${nompar}`);
  }

  //Valida codpar
  valCodpar(codpar: string): Observable<boolean> {
    const url = `${baseUrl}/valcodpar/${encodeURIComponent(codpar)}`;
    return this.http.get<boolean>(url);
  }

  //Valida nompar
  valNompar(nompar: string): Observable<boolean> {
    const url = `${baseUrl}/valnompar/${encodeURIComponent(nompar)}`;
    return this.http.get<boolean>(url);
  }

  //Partidas del clasificador por codpar para datalist
  findPartidasing(codpar: string) {
    // console.log(`${baseUrl}/partidasing?codpar=${codpar}`)
    return this.http.get<Clasificador[]>(`${baseUrl}/partidasing?codpar=${codpar}`);
  }

  //Guarda nuevo
  save(partida: Clasificador): Observable<Object> {
    return this.http.post(baseUrl, partida);
  }

  //Busca por idclasificador
  getByIdclasificador(idclasificador: number) {
    return this.http.get<Clasificador>(baseUrl + "/" + idclasificador);
  }

  //Modifica
  update(idclasificador: number, partida: Clasificador): Observable<Object> {
    return this.http.put(baseUrl + "/" + idclasificador, partida);
  }

  //Elimina
  delete(idclasificador: number) {
    return this.http.delete(`${baseUrl}/${idclasificador}`);
  }

}